Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1654,38 +1774,38 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes(GridLayoutState& gridL
 
 void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizesMasonry(GridLayoutState& gridLayoutState)
 {
-    if (m_strategy->isComputingSizeContainment()) {
+    if (m_strategy->isComputingSizeContainment() || !m_grid.hasGridItems()) {
         handleInfinityGrowthLimit();
         return;
     }
+    StdMap<SpanLength, MasonryMinMaxTrackSize> indefiniteItems;
+    StdMap<SpanLength, Vector<MasonryMinMaxTrackSizeWithGridSpan>> definiteItems;
+    Vector<MasonryMinMaxTrackSizeWithGridSpan> definiteItemsSpanFlexTrack;
```

Comment:
Personally not a fan of how these structures are named as holding "items," but they don't actually hold items but instead sizes computed in a certain context